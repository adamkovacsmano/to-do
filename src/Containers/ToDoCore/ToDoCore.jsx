import React, { Component } from "react";
import styles from "./ToDoCore.module.scss";
import ListItem from "../../Components/ListItem";
import { firestore } from "../../firebase";
import Button from "../../Components/Button";

class ToDoCore extends Component {
  state = {
    list: [],
    filteredList: [],
    textInput: "",
    userName: this.props.userName,
    userImage: this.props.userImage
  };

  renderListItems = () => {
    firestore
      .collection("listItems")
      .get()
      .then(querySnapshot => {
        const list = querySnapshot.docs.map(doc => {
          return { ...doc.data(), docId: doc.id };
        });
        this.setState({
          list: list,
          filteredList: list
        });
      });
  };
  componentDidMount = () => {
    this.renderListItems();
    document.addEventListener("keydown", this.addListItem, false);
  };

  addListItem = event => {
    if (event.keyCode === 13) {
      let data = {
        text: this.state.textInput,
        isMarked: false,
        userName: this.state.userName,
        userImage: this.state.userImage
      };
      firestore
        .collection("listItems")
        .add(data)
        .then(() => this.renderListItems())
        .catch(console.log("hups"));
    }
  };

  deleteItem = docId => {
    firestore
      .collection("listItems")
      .doc(docId)
      .delete()
      .then(this.renderListItems);
  };

  markAsDone = docId => {
    firestore
      .collection("listItems")
      .doc(docId)
      .update({ isMarked: true })
      .then(this.renderListItems);
  };

  updateItem = docId => {
    firestore
      .collection("listItems")
      .doc(docId)
      .update({ textInput: "" })
      .then(this.renderListItems);
  };

  getTextValue = event => {
    this.setState({ textInput: event.target.value });
  };

  handleChange = event => {
    let items = this.state.list;
    items = items.filter(item => {
      return (
        item.text.toLowerCase().search(event.target.value.toLowerCase()) !== -1
      );
    });
    this.setState({ items: items, filteredList: items });
  };

  handleClick = event => {
    const filteredList = this.state.list.filter(item => {
      if (event.target.innerHTML === "Done") {
        return item.isMarked === true;
      } else if (event.target.innerHTML === "ToDo") {
        return item.isMarked === false;
      } else {
        return event.target.innerHTML === "Full list";
      }
    });
    this.setState({ filteredList: filteredList });
  };

  render() {
    return (
      <div className={styles.container}>
        <input
          id="testInput"
          type="text"
          placeholder="..add your stuff"
          onKeyPress={this.addListItem}
          onChange={this.getTextValue}
          value={this.state.textInput}
        ></input>
        <input
          type="text"
          placeholder="..search"
          onChange={this.handleChange}
        ></input>
        <div className={styles.buttonContainer}>
          <Button
            id="testButton"
            name={"Done"}
            onClick={this.handleClick}
          ></Button>
          <Button
            id="testButton2"
            name={"ToDo"}
            onClick={this.handleClick}
          ></Button>
          <Button
            id="testButton3"
            name={"Full list"}
            onClick={this.handleClick}
          ></Button>
        </div>
        <p>click on items to mark them as done</p>
        <ul>
          {this.state.filteredList.map(item => (
            <ListItem
              markAsDone={this.markAsDone}
              deleteItem={this.deleteItem}
              updateItem={this.updateItem}
              text={item.text}
              key={item.docId}
              data={item}
              isMarked={item.isMarked}
              userName={item.userName}
              userImage={item.userImage}
            ></ListItem>
          ))}
        </ul>
      </div>
    );
  }
}

export default ToDoCore;
