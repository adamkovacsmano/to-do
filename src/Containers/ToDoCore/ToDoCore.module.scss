.container {
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%);
  display: flex;
  flex-direction: column;
  align-items: center;

  h1 {
    text-align: center;
  }

  p {
    font-style: italic;
  }

  input {
    margin-bottom: 10px;
    width: 100%;
  }
}

.buttonContainer {
  display: flex;
  justify-content: space-around;
  width: 100%;
}

@media screen and (max-width: 600px) {
  .container {
    width: 100%;
    input {
      width: 300px;
    }
  }

  .buttonContainer {
    width: 300px;
  }
}
