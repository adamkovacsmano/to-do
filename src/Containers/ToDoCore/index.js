import ToDoCore from "./ToDoCore";

export default ToDoCore;
