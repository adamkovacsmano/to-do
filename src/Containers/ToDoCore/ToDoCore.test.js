import React from "react";
import { shallow, mount } from "enzyme";
import ToDoCore from "./ToDoCore";

describe("ToDoCore", () => {
  it('should render correctly in "debug" mode', () => {
    const todocore = shallow(<ToDoCore debug />);

    expect(todocore).toMatchSnapshot();
  });
});

it("should be possible to press Enter on textarea", () => {
  const component = mount(<ToDoCore />);
  component.find("input#testInput").simulate("keydown", { keyCode: 13 });
  expect(component).toMatchSnapshot();
  component.unmount();
});

it("should be possible to press Enter on textarea", () => {
  const component = mount(<ToDoCore />);
  component.find("input#testInput").simulate("change");
  expect(component).toMatchSnapshot();
  component.unmount();
});

it("should be possible to activate Button with onClick", () => {
  const component = mount(<ToDoCore />);
  component.find("Button#testButton").simulate("click");
  expect(component).toMatchSnapshot();
  component.unmount();
});

it("should be possible to activate Button with onClick", () => {
  const component = mount(<ToDoCore />);
  component.find("Button#testButton2").simulate("click");
  expect(component).toMatchSnapshot();
  component.unmount();
});

it("should be possible to activate Button with onClick", () => {
  const component = mount(<ToDoCore />);
  component.find("Button#testButton3").simulate("click");
  expect(component).toMatchSnapshot();
  component.unmount();
});

// --------- Expected number of calls: >= 1 not: Received number of calls:    0

// const mockFunction = jest.fn();
// it("should call mockFunction on button click", () => {
//   const component = mount(<ToDoCore onClickFunction={mockFunction} />);
//   component.find("Button#testButton").simulate("click");
//   expect(mockFunction).toHaveBeenCalled();

//   component.unmount();
// });
