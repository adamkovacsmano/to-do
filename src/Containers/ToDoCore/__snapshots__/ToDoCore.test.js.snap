// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoCore should render correctly in "debug" mode 1`] = `
<div
  className="container"
>
  <input
    id="testInput"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="..add your stuff"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="..search"
    type="text"
  />
  <div
    className="buttonContainer"
  >
    <Button
      id="testButton"
      name="Done"
      onClick={[Function]}
    />
    <Button
      id="testButton2"
      name="ToDo"
      onClick={[Function]}
    />
    <Button
      id="testButton3"
      name="Full list"
      onClick={[Function]}
    />
  </div>
  <p>
    click on items to mark them as done
  </p>
  <ul />
</div>
`;

exports[`should be possible to activate Button with onClick 1`] = `
<ToDoCore>
  <div
    className="container"
  >
    <input
      id="testInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="..add your stuff"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="..search"
      type="text"
    />
    <div
      className="buttonContainer"
    >
      <Button
        id="testButton"
        name="Done"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <Button
        id="testButton2"
        name="ToDo"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          ToDo
        </button>
      </Button>
      <Button
        id="testButton3"
        name="Full list"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Full list
        </button>
      </Button>
    </div>
    <p>
      click on items to mark them as done
    </p>
    <ul />
  </div>
</ToDoCore>
`;

exports[`should be possible to activate Button with onClick 2`] = `
<ToDoCore>
  <div
    className="container"
  >
    <input
      id="testInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="..add your stuff"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="..search"
      type="text"
    />
    <div
      className="buttonContainer"
    >
      <Button
        id="testButton"
        name="Done"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <Button
        id="testButton2"
        name="ToDo"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          ToDo
        </button>
      </Button>
      <Button
        id="testButton3"
        name="Full list"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Full list
        </button>
      </Button>
    </div>
    <p>
      click on items to mark them as done
    </p>
    <ul />
  </div>
</ToDoCore>
`;

exports[`should be possible to activate Button with onClick 3`] = `
<ToDoCore>
  <div
    className="container"
  >
    <input
      id="testInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="..add your stuff"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="..search"
      type="text"
    />
    <div
      className="buttonContainer"
    >
      <Button
        id="testButton"
        name="Done"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <Button
        id="testButton2"
        name="ToDo"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          ToDo
        </button>
      </Button>
      <Button
        id="testButton3"
        name="Full list"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Full list
        </button>
      </Button>
    </div>
    <p>
      click on items to mark them as done
    </p>
    <ul />
  </div>
</ToDoCore>
`;

exports[`should be possible to press Enter on textarea 1`] = `
<ToDoCore>
  <div
    className="container"
  >
    <input
      id="testInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="..add your stuff"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="..search"
      type="text"
    />
    <div
      className="buttonContainer"
    >
      <Button
        id="testButton"
        name="Done"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <Button
        id="testButton2"
        name="ToDo"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          ToDo
        </button>
      </Button>
      <Button
        id="testButton3"
        name="Full list"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Full list
        </button>
      </Button>
    </div>
    <p>
      click on items to mark them as done
    </p>
    <ul />
  </div>
</ToDoCore>
`;

exports[`should be possible to press Enter on textarea 2`] = `
<ToDoCore>
  <div
    className="container"
  >
    <input
      id="testInput"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="..add your stuff"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="..search"
      type="text"
    />
    <div
      className="buttonContainer"
    >
      <Button
        id="testButton"
        name="Done"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Done
        </button>
      </Button>
      <Button
        id="testButton2"
        name="ToDo"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          ToDo
        </button>
      </Button>
      <Button
        id="testButton3"
        name="Full list"
        onClick={[Function]}
      >
        <button
          className="primary"
          id="testButton2"
          onClick={[Function]}
        >
          Full list
        </button>
      </Button>
    </div>
    <p>
      click on items to mark them as done
    </p>
    <ul />
  </div>
</ToDoCore>
`;
